Implement PSR-11

MauricioFauth opened the pull request:
Fixes #13052

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [ ] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

MauricioFauth committed:
Implement PSR-11

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

MauricioFauth committed:
Fix trailing whitespace

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

MauricioFauth committed:
Add ContainerTest

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

mauriciofauth commented:
I've never written tests before, so I do not know if the ones I've written are correct.

MauricioFauth committed:
Remove unused classes [skip ci]

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

nijel commented on libraries/di/Container.php:
```diff
@@ -46,19 +47,58 @@ public function __construct(Container $base = null)
+        } catch (NotFoundException $e) {
+            echo $e->getMessage();
+            return null;
+        }
```
I really don't get the code here:

- Why is there `try { throw } catch {}` block at all?
- The echo is there for something else than debugging purposes?

nijel commented on libraries/di/Container.php:
```diff
@@ -46,19 +47,58 @@ public function __construct(Container $base = null)
      * @param array  $params Parameters
      *
+     * @throws NotFoundException  No entry was found for **this** identifier.
+     * @throws ContainerException Error while retrieving the entry.
```
It doesn't seem to throw these as they are always handled inside...

nijel commented on libraries/di/Container.php:
```diff
@@ -46,19 +47,58 @@ public function __construct(Container $base = null)
         }
 
-        return null;
+        return false;
```
Probably better to write whole method as:

```php
    return isset($this->content[$name]) || isset($GLOBALS[$name]);
```

nijel commented on test/classes/di/ContainerTest.php:
```diff
@@ -0,0 +1,93 @@
+     */
+    public function testGetWithInvalidEntry()
+    {
+        $this->assertSame(null, $this->container->get('invalid'));
```
IMHO it should throw NotFoundExceptionInterface according to PSR-11.

MauricioFauth committed:
Apply requested changes and add more tests

Remove try-catch blocks
Refactor has method
Refactor testGetWithInvalidEntry method
Add tests for ContainerException and NotFoundException

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

nijel commented:
And now the code code which uses container needs to be adjusted to understand new behavior, see at least the test failures: https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/210553712#L457

mauriciofauth commented:
I will do this @nijel 

And about the `get` method, I think it would not be appropriate to use `isset` since this already happens in the `has` method. I think it should check whether the value is valid or not. Maybe `!is_null`?

nijel commented:
Why should it check for null value? I think it's perfectly okay to return null if it has been set...

mauriciofauth commented:
Oh, of course. I was not thinking straight.
Looking at the code, I realized that all values are instances of `ValueItem`. Maybe check for it.
I'm wanting to check for something other than `isset`, otherwise `ContainerException` will never be thrown.

MauricioFauth committed:
Fix uncaught NotFoundException

Signed-off-by: Maurício Meneghini Fauth <mauriciofauth@gmail.com>

mauriciofauth commented:
I left `get` method the way it was and corrected the tests that were failing.
Is everything all right now, @nijel?

nijel commented:
Merged, thanks for your contribution!
